plugin/adsk/plugin/mayaUsdInfoCommand.cpp

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright 2024 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "mayaUsdInfoCommand.h"
+
+#include <mayaUsd/buildInfo.h>
+
+#include <maya/MArgParser.h>
+#include <maya/MSyntax.h>
+
+#define XSTR(x) STR(x)
+#define STR(x)  #x
+
+namespace MAYAUSD_NS_DEF {
+
+const MString MayaUsdInfoCommand::commandName("mayaUsd");
+
+namespace {
+
+// Versioning and build information.
+constexpr auto kMajorVersion = "-mjv";
+constexpr auto kMajorVersionLong = "-majorVersion";
+
+constexpr auto kMinorVersion = "-mnv";
+constexpr auto kMinorVersionLong = "-minorVersion";
+
+constexpr auto kPatchVersion = "-pv";
+constexpr auto kPatchVersionLong = "-patchVersion";
+
+constexpr auto kVersion = "-v";
+constexpr auto kVersionLong = "-version";
+
+constexpr auto kCutId = "-c";
```

Comment:
I don't think so. The cutids are not internal. Its just a combination of the build date and commit sha. I discovered that the branch name is actually not very useful since it will usually be `master`. I'm going to look into possibly changing this one.